### DISCUSSION TITLE
STM32F407 DownloadFile function throw exception

### DISCUSSION THREAD

**Discussion by reza471** [K-Society/KSociety.SharpCubeProgrammer#101] [Q&A]
Hi

**Comment by maniglia** [comment]
Hello @reza471,
Please can you provide me with some more details.
What kind of problem? 
An exception?
On which platform, x64 or x86?
Regards

**Comment by reza471** [comment]
Hi
thank for for your api. 
I previously develop in c++.net and work fine.
I get Error Memory Program in Download file   
In small file does not error  but in bigger i got error  100kb and  1Mb test
Stm32f407zg
x64
.netcore5

**Reply by maniglia** [reply]
Hello @reza471 , 
I've never tested the download feature with large files. It's very likely that you've found a bug. I'll try to reproduce the problem when I find the time to do so, in the meantime I ask you to update the firmware of your STLINK and try again.

Regards

**Reply by reza471** [reply]
![bug](https://github.com/K-Society/KSociety.SharpCubeProgrammer/assets/12105672/b8b1e387-31f1-4ac9-9f46-c4b0f8db1527)
Using 700kb  is OK  , but on 2.435kb  have error

**Reply by maniglia** [reply]
Hello @reza471 
I just ran a test with the _DownloadFile_ function with a 1MB _bin_ file on _x86_ an _x64_ platform (I'm using the **NUCLEO-WBA52CG**).
All tests gave positive results without any exceptions. 
However, I am using a version of the library that I have not yet published.
Please you can try **not** to use the _SetDisplayCallbacks_ function.
Let me know thanks.

**Reply by reza471** [reply]
Thank you for the help offer
I think ,problem related to  the SetDisplayCallbacks function.
but every programming , i need to Disconnect and Connect again.

**Reply by maniglia** [reply]
There would be the _AutomaticMode_ function for sequential programming which however I have never tested yet.

**Comment by maniglia** [comment]
Hello @reza471 ,
there is a new version of the library, 0.2.4.
The problems you reported should now no longer arise, I hope :)
Regards

**Reply by reza471** [reply]
Thanks
